lua/entities/lucian_door_opener.lua

```diff
@@ -36,108 +38,128 @@ ENT.SpritePositions = {
+	if CLIENT then
+		for _,device in pairs(tab) do
+			device.SpriteColor = color
+		end
+	else
+		net.Start("CAP_LDO_SpriteColorUpdate")
+		net.WriteTable(tab)
+		net.WriteColor(color)
+		net.Broadcast()
+	end
+end
+
+if CLIENT then
+	net.Receive( "CAP_LDO_SpriteColorUpdate", function()
+		local tab = net.ReadTable()
+		local color = net.ReadColor()
+		for _,device in pairs(tab) do
+			if device != NULL then
+				device.SpriteColor = color
+			end
+		end
+	end)
+end
+
 if SERVER then
 
 if (StarGate==nil or StarGate.CheckModule==nil or not StarGate.CheckModule("devices")) then return end
 
 AddCSLuaFile();
 
 function ENT:Initialize()
-	self.Entity:SetModel("models/gmod4phun/lucian_door_opener.mdl");
-	self.Entity:PhysicsInit(SOLID_VPHYSICS);
-	self.Entity:SetMoveType(MOVETYPE_VPHYSICS);
-	self.Entity:SetSolid(SOLID_VPHYSICS);
-	self:SetUseType(ONOFF_USE)
-	self.Entity.Attached = false
-	self.Entity.SecondAttached = false
-	self.Entity.TargetDoor = nil
-	self.Entity.IsMain = nil
-	self.CanUse = CurTime()
-	self.GetTimer = 0
+	self:SetModel("models/gmod4phun/lucian_door_opener.mdl");
+	self:PhysicsInit(SOLID_VPHYSICS);
+	self:SetMoveType(MOVETYPE_VPHYSICS);
+	self:SetSolid(SOLID_VPHYSICS);
+	self:SetUseType(SIMPLE_USE)
+	self.Attached = false
+	self.SecondAttached = false
+	self.TargetDoor = nil
+	self.IsMain = nil
+	self.CanUse = true
 	self.Activator = NULL
 end
-/*
-function ENT:SpawnFunction( ply, tr )
-	if ( !tr.Hit ) then return end
-
-	local ang = ply:GetAimVector():Angle(); ang.p = 0; ang.r = 0; ang.y = (ang.y+180) % 360;
-
-	local ent = ents.Create("lucian_door_opener");
-	ent:SetAngles(ang);
-	ent:SetPos(tr.HitPos);
-	ent:Spawn();
-	ent:Activate();
-
-	local phys = ent:GetPhysicsObject()
-	if IsValid(phys) then phys:EnableMotion(true) end
-
-	return ent
-end*/
 
 function ENT:RemoveBothLDO()
-	if IsValid(self.Entity) then
-		self.Entity:Remove()
+	if IsValid(self) then
+		self:Remove()
 	end
-	if IsValid(self.Entity) and IsValid(self.Entity.SecondLucianDevice) then
-		self.Entity.SecondLucianDevice:Remove()
+	if IsValid(self) and IsValid(self.SecondLucianDevice) then
+		self.SecondLucianDevice:Remove()
 	end
-	if IsValid(self.Entity) and IsValid(self.Entity.MainLDO) then
-		self.Entity.MainLDO:Remove()
+	if IsValid(self) and IsValid(self.MainLDO) then
+		self.MainLDO:Remove()
 	end
 end
 
-function ENT:Use(activator,_,use)
-	
-	if self.CanUse>CurTime() then
-		if (use==0) then
-			self.GetTimer = 0
-			self.Activator = NULL
-		end
-		return
-	end
+function ENT:Use(activator)
 
-	if not IsValid(self.TargetDoor) then return end
-	
-	self.CanUse = CurTime()+2.5
-	self.GetTimer = CurTime()+0.75
-	self.Activator = activator
+	if !self.CanUse then return end
 	
-	/*if IsValid(self.Entity) and IsValid(self.Entity.TargetDoor) and activator:KeyDown(IN_WALK) then
+	if not IsValid(self.TargetDoor) then return end
+
+	if IsValid(self) and IsValid(self.TargetDoor) and activator:KeyDown(IN_WALK) then -- Let us pick up when holding ALT and press E
```

Comment:
I'm nor sure what is better, hold E or press ALT + E, really, there is no much people will understand that you also need press ALT. Thats why i did it like hold - its easier to understand.